src/main.rs

```diff
@@ -119,22 +125,51 @@ fn work(args: Args) -> Result<(), Error> {
         }
     }
 
-    let license = if must_copy_license(&config.package.license) {
-        p("LICENSE file will be installed manually.".bold().yellow());
-        Some(license_file()?)
-    } else {
-        None
-    };
-
     if args.dryrun.not() {
-        release_build(args.musl)?;
-        tarball(args.musl, &cargo_target, &output, license.as_ref(), &config)?;
-        let sha256: String = sha256sum(&config.package, &output)?;
+        let (sha256, license) = match (args.crates, args.no_bin) {
+            (true, true) => {
+                let crate_file = crates::CrateFile::download_new(&config)?;
+                let sha256 = crate_file.get_sha256sum()?;
+                let license = crate_file.get_license()?;
+                (sha256, license)
+            }
+            (true, false) => {
+                let crate_file = crates::CrateFile::download_new(&config)?;
+                let built_crate_file = crate_file.build(args.musl)?;
+                let license = built_crate_file.tarball(&cargo_target)?;
+                let sha256 = sha256sum(&config.package, &output)?;
+                (sha256, license)
+            }
+            (false, true) => {
+                source_tarball(&cargo_target, &output, &config)?;
+                let license = alert_if_must_copy_license(&config.package.license)
+                    .then(|| license_file(None))
+                    .transpose()?;
+                let sha256 = sha256sum(&config.package, &output)?;
+                (sha256, license)
+            }
```

Comment:
This variant seems to produce a PKGBUILD that refers to a source with a different filename than the one produced by the build/packaging process. See my other comment about a potential shift in strategy for this variant.

Reply:
I have resolved this now, apologies.